internal/resource/obzone/obzone_task.go

```diff
@@ -396,3 +401,40 @@ func (m *OBZoneManager) MountBackupVolume() tasktypes.TaskError {
 	}
 	return nil
 }
+
+func (m *OBZoneManager) DeleteLegacyOBServer() tasktypes.TaskError {
+	operationManager, err := m.getOceanbaseOperationManager()
+	if err != nil {
+		return errors.Wrapf(err, "OBZone %s get oceanbase operation manager", m.OBZone.Name)
+	}
+	allOBServers, err := operationManager.ListServers()
+	if err != nil {
+		return errors.Wrap(err, "List observers in oceanbase")
+	}
+	observerList, err := m.listOBServers()
+	if err != nil {
+		return errors.Wrap(err, "List observer crs")
+	}
+	for _, observer := range allOBServers {
+		// skip observers in different obzone
+		if observer.Zone != m.OBZone.Spec.Topology.Zone {
+			continue
+		}
+		found := false
+		for _, observerCR := range observerList.Items {
+			if observer.Ip == observerCR.Status.PodIp {
+				found = true
+			}
+		}
+		if !found {
+			err := operationManager.DeleteServer(&model.ServerInfo{
+				Ip:   observer.Ip,
+				Port: observer.Port,
+			})
+			if err != nil {
+				return errors.Wrapf(err, "Delete observer %s:%d", observer.Ip, observer.Port)
+			}
+		}
+	}
```

Comment:
Shall we wait for the out server to be deleted?

Reply:
I prefer to not wait the legacy observers to be deleted, because delete server is a long time background task, obcluster will be stuck in migration status for a long time, besides if the new obcluster is unable to hold the original units, we can scale observer directly.